data_steward/cdr_cleaner/cleaning_rules/remove_invalid_procedure_source_records.py

```diff
@@ -46,61 +47,113 @@
  SELECT
     concept_id
   FROM
-    `{project}.{dataset}.concept`
+    `{{project}}.{{dataset}}.concept`
   WHERE
     TRIM(concept_class_id) = 'CPT4 Modifier'
 )
-"""
-
-VALID_PROCEDURE_SOURCE_CONCEPT_IDS_QUERY = """
-SELECT * FROM `{project}.{dataset}.{table}` as p0
-WHERE NOT EXISTS (
-SELECT p1.procedure_occurrence_id FROM `{project}.{sandbox_dataset}.{intermediary_table}` AS p1
-WHERE p0.procedure_occurrence_id = p1.procedure_occurrence_id)
-"""
-
-
-def get_remove_invalid_procedure_source_queries(project_id, dataset_id,
-                                                sandbox_dataset_id):
-    """
-    runs the query which removes records that contain incorrect values in the procedure_source_concept_id field
-    invalid procedure_source_concept_ids are where it is not in the procedure domain and
-    procedure_concept_id is not standard in the procedure domain
-
-    :param project_id: Name of the project
-    :param dataset_id: Name of the dataset where the queries should be run
-    :param sandbox_dataset_id: Identifies the sandbox dataset to store rows 
-    #TODO use sandbox_dataset_id for CR
-    :return:
-    """
-    queries_list = []
-
-    # queries to sandbox
-    invalid_records = dict()
-    invalid_records[
-        cdr_consts.QUERY] = INVALID_PROCEDURE_SOURCE_CONCEPT_IDS_QUERY.format(
-            project=project_id,
-            dataset=dataset_id,
-            table=TABLE,
-            sandbox_dataset=sandbox_dataset_id,
-            intermediary_table=INTERMEDIARY_TABLE_NAME)
-    queries_list.append(invalid_records)
-
-    # queries to delete invalid procedure source records
-    valid_records = VALID_PROCEDURE_SOURCE_CONCEPT_IDS_QUERY.format(
-        project=project_id,
-        dataset=dataset_id,
-        table=TABLE,
-        sandbox_dataset=sandbox_dataset_id,
-        intermediary_table=INTERMEDIARY_TABLE_NAME)
-    queries_list.append({
-        clean_consts.QUERY: valid_records,
-        clean_consts.DESTINATION_TABLE: TABLE,
-        clean_consts.DESTINATION_DATASET: dataset_id,
-        clean_consts.DISPOSITION: bq_consts.WRITE_TRUNCATE
-    })
-
-    return queries_list
+""")
+
+DELETE_INVALID_PROCEDURE_SOURCE_CONCEPT_IDS_QUERY = JINJA_ENV.from_string("""
+DELETE 
+FROM 
+  `{{project}}.{{dataset}}.{{table}}` AS p0
+WHERE EXISTS 
+(
+  SELECT 
+    p1.procedure_occurrence_id 
+  FROM 
+    `{{project}}.{{sandbox_dataset}}.{{sandbox_table}}` AS p1
+  WHERE 
+    p0.procedure_occurrence_id = p1.procedure_occurrence_id
+)
+""")
```

Comment:
This query's WHERE clause can be simplified as something like this:

```
WHERE procedure_occurrence_id IN (
  SELECT DISTINCT procedure_occurrence_id
  FROM `{{project}}.{{sandbox_dataset}}.{{sandbox_table}}`
)
```

This way we also do not need to assign the alias `p0` and `p1` to each of the tables.